### ISSUE TITLE
SQL error on PostgreSQL

### ISSUE THREAD

**Issue by mudrd8mz** [clickviewapp/moodle-mod_clickview#2]
Attempting to add a new instance on PostgreSQL gives an error:

```
Error reading from database

More information about this error
Debug info: ERROR: invalid input syntax for type bigint: ""
SELECT * FROM m_clickview WHERE id = $1
[array (
0 => '',
)]
Error code: dmlreadexception
Stack trace:

    line 486 of /lib/dml/moodle_database.php: dml_read_exception thrown
    line 329 of /lib/dml/pgsql_native_moodle_database.php: call to moodle_database->query_end()
    line 977 of /lib/dml/pgsql_native_moodle_database.php: call to pgsql_native_moodle_database->query_end()
    line 1671 of /lib/dml/moodle_database.php: call to pgsql_native_moodle_database->get_records_sql()
    line 1643 of /lib/dml/moodle_database.php: call to moodle_database->get_record_sql()
    line 1622 of /lib/dml/moodle_database.php: call to moodle_database->get_record_select()
    line 167 of /mod/clickview/mod_form.php: call to moodle_database->get_record()
    line 83 of /mod/clickview/mod_form.php: call to mod_clickview_mod_form->get_thumbnail_html()
    line 214 of /lib/formslib.php: call to mod_clickview_mod_form->definition()
    line 130 of /course/moodleform_mod.php: call to moodleform->__construct()
    line 150 of /course/modedit.php: call to moodleform_mod->__construct()
```

**Comment by mudrd8mz** [comment]
On that note, why is there `string $id` typehint at https://github.com/clickviewapp/moodle-mod_clickview/blob/18f1f7415d7f264d0cb40f1ccb2ed5a6140d2a45/mod_form.php#L162 ? Identifiers are integers in Moodle database tables ...

**Comment by adpe** [comment]
Thanks @mudrd8mz 

It's a bug and should now be resolved. Unfortunately testing it out with Behat I've faced some problems due JS loading on the page.

Can you check out the branch https://github.com/clickviewapp/moodle-mod_clickview/tree/bugfix/2-add-activity and verify it your issue is solved?

Greets
Adrian

**Comment by mudrd8mz** [comment]
Were you able to confirm the fix via manual testing on Postres?

**Comment by adpe** [comment]
> Were you able to confirm the fix via manual testing on Postres?

Yep! Just set up a clean Moodle environment and was possible to reproduce your error. After checking out the patch, it is fixed. Did it not solve it on your end?

PS: But noticed the same time that you can add a new ClickView activity when no video is selected. This should not be what was expected. Shall I fix it @Shalelol?

Greets
Adrian

**Comment by mudrd8mz** [comment]
Great. Please make sure a new version with the fix is released in the Plugins directory as that is the only source we check for approval. We are not able to test individual fixes on github. Thank you for understanding.

**Comment by Shalelol** [comment]
Thanks for working on this @adpe. I've removed the behat test for the time being and created some new relases, I can look into getting that working properly later. In regards to this issue https://github.com/clickviewapp/moodle-mod_clickview/issues/3. I can look into fixing that myself and will reach out if I need help.

@mudrd8mz I've submitted the new versions to the Plugin directory and have requested approval.

Thanks you both for your help.
